core/native/src/main/scala/bobcats/HashPlatform.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bobcats
+
+import cats.effect.kernel.Async
+import scodec.bits.ByteVector
+import scalanative.unsafe._
+import scalanative.unsigned._
+import openssl._
+import openssl.err._
+import openssl.evp._
+
+private[bobcats] trait HashCompanionPlatform {
+  implicit def forAsync[F[_]](implicit F: Async[F]): Hash[F] =
+    new UnsealedHash[F] {
+      override def digest(algorithm: HashAlgorithm, data: ByteVector): F[ByteVector] = {
+
+        import HashAlgorithm._
+
+        val digest = algorithm match {
+          case MD5 => EVP_md5()
```

Comment:
Semantically, I need to double check whether the symbols are still present if you compile your openssl without the algorithms, or this can return null.

Reply:
You can't seem to disable these constants. The `InitX` code is null-safe regardless.